File: src/test/fixtures/typescript/coherence/boundCallables.ts
function encode(value: number): number {
    return value;
}

const moduleBound = (value: string): string => value;

function outer(items: boolean[]): boolean[] {
    const localBound = (value: boolean): boolean => value;
    return items.map(item => item);
}

class Operations {
    classBound = (value: Date): Date => value;

    namedMethod(value: RegExp): Promise<void> {
        return Promise.resolve();
    }

    outerMethod(items: Set<number>): Map<number, number> {
        const localBound = (value: number): number => value;
        return new Map(Array.from(items, item => [item, item]));
    }
}

File: src/test/fixtures/python/coherence/bound_callables.py
def encode(value):
    return value


module_bound = lambda value: value


def outer(items):
    local_bound = lambda value: value
    return list(map(lambda item: item, items))


class Operations:
    class_bound = lambda value: value

    def named_method(self, value: int) -> str:
        return str(value)

    def outer_method(self, items: list) -> tuple:
        local_bound = lambda value: value
        return tuple(map(lambda item: item, items))
